### ISSUE TITLE
GameStudioThumbnailService.cs Debugger diagnostics break

### ISSUE THREAD

**Issue by avestura** [stride3d/stride#27]
```
if (thumbnailsInProgressAndContinuation.ContainsKey(asset.Id) && System.Diagnostics.Debugger.IsAttached)
{
    // Virgile: This case should not happen, but it happened to me once and could not reproduce.
    // Please let me know if it happens to you.
    // Note: this is likely not critical and should work fine even if it happens.
    System.Diagnostics.Debugger.Break();
}
```

> Virgile: This case should not happen,...

Anything that can go wrong, will go wrong. -- Morphy Law

**Comment by xen2** [comment]
Yes it was happening once in a while, always seemed harmless but will have to check the logic details again someday.

**Comment by NicusorN5** [comment]
Why was this closed? Is this reproductible anymore?

**Comment by avestura** [comment]
> Why was this closed? Is this reproductible anymore?

@NicusorN5  Well, it appears the last time it happened, or anyone faced an issue related to this, was 7 years ago. I thought we might be able to close this and reopen it later when it happens again.

**Comment by MsEpsilon** [comment]
Reproduced yesterday by opening the same project twice.

**Comment by IXLLEGACYIXL** [comment]
> Reproduced yesterday by opening the same project twice. The asset was from `Stride.Video`.

the editor crashed on first opening?

**Comment by MsEpsilon** [comment]
No, but the project being loaded had compile errors (if I remember well, I already forgot)
